presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1637,6 +1637,40 @@ private Scope computeAndAssignOrderByScope(OrderBy node, Scope sourceScope, Scop
             return orderByScope;
         }
 
+        private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope outputScope, List<FunctionCall> aggregations, List<List<Expression>> groupByExpressions)
+        {
+            // When aggregation is present then FROM (non-aggregated) fields should be hidden for ORDER BY during planning.
```

Comment:
hidden or just forbidden? I.e. in 
```
select (
    select b
    from (values (1, 2)) inn(a, b)
    group by b
    order by a -- sic!
)
from (values (1)) out(a);
```

we need to bind `a` in `order by` as `inn.a` (and reject) rather than bind as `out.a` (and accept).

Even if the code works now (I don't know), do you have tests covering this?